### DISCUSSION TITLE
About sampling rate?

### DISCUSSION THREAD

**Discussion by 1248280302** [aewallin/allantools#140] [General]
How do I set the parameters if I want to enter satellite atomic clock data at 5 minute intervals.
For example this function：
(adev_taus,adev_devs,adev_errs,ns)  = allan.adev(phase, rate=rate, taus=my_taus)
I try to input 151200 data but the final output calculation result is much less than this.
What is the role of this logspace:
my_taus = numpy.logspace(1,6,60)
Should I set the rate parameter like
rate = 1/float(300.0)

**Comment by aewallin** [comment]
Your rate = 1/float(300.0)  looks correct if you have input data at 5*60s = 300s intervals.

if you have len(phase) = 151200 you will be able to calculate adev for a maximum tau of about (len(phase)/2)*300 = 22680000 s

For the taus input-parameter you could also use 'octave' or 'decade' or 'all' (this last one could be slow, with large amounts of input data) as described here: https://allantools.readthedocs.io/en/latest/functions.html#low-level-access-to-the-statistics-functions

hope this helps.

**Comment by 1248280302** [comment]
@aewallin Your suggestion is useful, I configure it according to the parameters you said
rate = 1 / float(300.0)
my_taus = np.arange(1, len(phase), step=300)
I get the result numerically close to correct,but the curve shape is not quite right.
My Harmard dev curve is looks like the ‘√’ sign,but the normal result should be more similar to the result after the horizontal flip.
Could it be a problem with setting parameters?

**Comment by aewallin** [comment]
I don't know. Can you post a minimal example with your input data, or simulated data, and the results you get.

**Comment by 1248280302** [comment]
@aewallin Thanks,I've got the correct result, 151200 data can't be fed into the calculation all at once, after splitting it into shorter data it works fine.
This program is powerful, thanks again!
